### PR TITLE
[oneMKL][spblas] Add sorted_by_rows property

### PR DESCRIPTION
**Pull request by Rbiessy**
Following the discussion in https://github.com/oneapi-src/oneMKL/pull/527#discussion_r1666587321 I suggest to introduce the `sorted_by_rows` property.

Note that this [section](https://github.com/uxlfoundation/oneAPI-spec/blob/main/source/elements/oneMKL/source/domains/spblas/data_types/format-descriptions.rst?plain=1#L29) already specifies that the default state of COO format is unsorted.

See compiled html page: [html.zip](https://github.com/user-attachments/files/17101211/html.zip)

### CODE REVIEW

File: source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst [hunk 1]
```diff
@@ -39,12 +40,20 @@ Matrix properties
...
             column in ascending order.
+      * - ``sorted_by_rows``
+        - | Guarantees that the user-provided matrix data is partially sorted.
+            Has no effect if the property ``sorted`` is set.
+          | For COO this guarantees that the indices are sorted by row in
+            ascending order. The column indices within a row do not need to be
+            sorted.
+          | Setting this property for a matrix using the CSR format will throw an
+            :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+            exception.
```
Comment (gajanan-choudhary) [inline source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst:56]: Why should it throw an `invalid_argument` exception for CSR format? This is the default state of a CSR format anyway. Setting this property on a CSR matrix should do nothing regardless of whether the `sorted` property has been set or not.

Comment (Rbiessy) [reply source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst:56]: I don't really think of the CSR as being sorted by rows in the sense that the values in `row_ptr` don't represent indices and can't be ordered in any other way. I understand the idea for allowing this. I think not throwing an exception is a risk for inexperienced users that maybe wanted to set the `sorted` property instead.

Comment (gajanan-choudhary) [reply source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst:56]: > I think not throwing an exception is a risk for inexperienced users that maybe wanted to set the sorted property instead.

Good thinking. This is fine then.

**Comment by Rbiessy** [comment]
Update the table to fix https://github.com/uxlfoundation/oneAPI-spec/issues/588

New html doc: [docs.zip](https://github.com/user-attachments/files/17151858/docs.zip)

File: source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst [hunk 2]
```diff
@@ -28,23 +29,51 @@ Matrix properties
...
+      :widths: 20 80
+
+      * - Matrix format
+        - Description
+      * - CSR
+        - Guarantees that the column indices are sorted in ascending order for
+          a given row.
+      * - COO
+        - Guarantees that the indices are sorted by row then by column in
+          ascending order.
```
Comment (gajanan-choudhary) [inline source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst:56]:
```suggestion
        - Guarantees that the indices are sorted by rows then by columns
          within rows in ascending order.
```

**Review by gajanan-choudhary — APPROVED** [review]
LGTM

File: source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst (hunk 2 above)
Comment (Rbiessy) [reply source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst]: Done in https://github.com/uxlfoundation/oneAPI-spec/pull/583/commits/eef74a30ab15edbfed9e1d470913aec66ec23903

**Pull request merged**